agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -37,6 +39,10 @@ def ensure_desired_state
       exclusive {
         debug "state of #{service_pod.name}: #{service_pod.desired_state}"
         service_container = get_container(service_pod.service_id, service_pod.instance_number)
+        if service_container && !service_pod.running?
```

Comment:
These conditions only apply to healthcheck, so maybe it would make sense to do them in `stop_health_check`?